packages/medusa/src/services/cache.ts

```diff
@@ -0,0 +1,43 @@
+import { Redis } from "ioredis"
+
+const DEFAULT_CACHE_TIME = 60 * 60 // 60 seconds
+
+export default class CacheService {
```

Comment:
praise: This is great!

thought: This couples us to Redis as caching layer. We could leverage our new Module Resolution API to create a CacheAbstraction (similar to the one @carlos-r-l-rodrigues proposed) as well as a module definition. The module could use the Redis service as a default.

Reply:
In fact, it abstract redis to a single place instead of having redis reference in the entire code base as it is at the moment. If all the places use this service instead of redis directly it means that first it can become a module/strategy and second that if we want to use something else, we just have to change the implementation of the cache service method and it will be applied on the entire code base.

Reply:
Not sure I understand.

My idea was to create an `ICacheService` interface exposing three methods; set, get, invalidate - in the first iteration.

Then create a module `@medusajs/cache-redis` similar to the event bus + inventory management proposals, and use that module as default in the module definition. 

That would allow developers to use any cache service, create a module for it, and replace the default Redis one.

Reply:
Maybe you would prefer that I remove the cache service (which is an intermediary step toward the module) and instead inject the `redisClient` directly in the price selection strategy which means that we coupled another place to it

Reply:
Since this is not part of that pr,  I have just created the appropriate interface that the cache service implements. In another pr we will be able to extract it to a separate module as it was the intention behind what I have done but I wanted to keep it simple in that pr. would you like me to go further

Reply:
Maybe it would be a good idea to let the cache out of this PR completely. You can keep a few `// TODO:` comments to mark the places where we could cache the values.
Ideally we should implement a more robust solution for caching things and do not add much complexity in the code itself with many if/else to check whether it has cache and set it manually.

Reply:
yeah that is true, but I need the cache here otherwise I get an improvement of the perf but not enough. Also, this is a place where it is required. My idea was to not complexity to much the price selection strategy and try to not reference anything related to redis and therefore extract that to a service.

Reply:
My suggestion was to create a ticket after that PR to extract the cache system. But we can't not use it in the mean time. So this is why I tried to find an intermediary steps where we remove the notion of redis from the services to only have it in one place (the cache service) and therefore simplify the migration later. But in the mean time it still allow us to use it without adding more reference to redis itself into the services

Reply:
wdyt @carlos-r-l-rodrigues

Reply:
I don't have enough knowledge about all the moving pieces composing the price selection, but at a first glance I find the cache key very granular:
`ps:${variantId}:${context.region_id}:${context.currency_code}:${context.customer_id}:${context.quantity}:${context.include_discount_prices}:${taxRate}`

I'm affraid that we will create a gigantic amount of entries if many users are using the system and that will create other type of problems.

Reply:
At the moment, if a number of concurrent users are doing it without the cache it causes a lot of problem as well as this will be called a gigantic amount of time as well. I think it is a trade off and i ve benchmarked it with a load test as well and it is much better

Reply:
After a discussion with @carlos-r-l-rodrigues we agreed upon the trade off as a temporary solution until the new approach a cache system is available.